staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go

```diff
@@ -190,14 +190,18 @@ type LeaderElector struct {
 func (le *LeaderElector) Run(ctx context.Context) {
 	defer func() {
 		runtime.HandleCrash()
-		le.config.Callbacks.OnStoppedLeading()
+		if le.config.Callbacks.OnStoppedLeading != nil {
+			le.config.Callbacks.OnStoppedLeading()
+		}
 	}()
 	if !le.acquire(ctx) {
 		return // ctx signalled done
 	}
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
-	go le.config.Callbacks.OnStartedLeading(ctx)
+	if le.config.Callbacks.OnStartedLeading != nil {
```

Comment:
The other one is fine, but if this is nil, we should probably panic. There's no point to this client if you don't do anything when you get leadership.

Reply:
A client could potentially not actively want to be notified of it being a leader and but choose to check the leader status using `IsLeader()` function whenever it is actually going to take up any action.

Reply:
You can do this by providing an empty OnStartedLeading callback. I am hesitant to further facilitate this behavior.

Reply:
Mike has a good point. That having been said I personally would rather an explicit panic which explained the issue (eg. "Leader elect called without a proper started leading method") than a fatal error about referencing invalid memory.

Reply:
>  You can do this by providing an empty OnStartedLeading callback. I am hesitant to further facilitate this behavior.

My take is that zero values in Go shouldn't panic in such a case.

Please let me know the exact change that should be made to move this forward. You guys are more familiar with the API style and end-user expectations here so I'll make a change that works for everyone.